source/agora/node/Ledger.d

```diff
@@ -882,6 +904,26 @@ public class Ledger
         return this.slash_man.getExternalizedRandomSeed(height,
             missing_validators);
     }
+
+    /***************************************************************************
+
+        Calculate and accumulate fees that will be paid to Validators from this
+        block
+
+        Params:
+            block = new block
+
+    ***************************************************************************/
+
+    public void accumulateFees (const ref Block block) nothrow @safe
+    {
+        UTXO[] stakes;
+        this.enroll_man.getValidatorStakes(&this.utxo_set.peekUTXO, stakes);
+        assert(isStrictlyMonotonic(block.header.missing_validators));
+        foreach (idx; block.header.missing_validators.retro)
+            stakes = stakes.remove(idx);
```

Comment:
Wouldn't it be more efficient to provide these missing validators in the `getValidatorStakes` call?

I guess it doesn't matter much if slashing will be rare though.

Reply:
Yep, I'll do that. Performance benefits may be negligible but it tidies up the caller site. Thanks!